File: ExampleApp/src/PlayerCube.cs
using Prism;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Example
{
    class PlayerCube : Behaviour
    {
        public float HorizontalForce = 10.0f;
        public float JumpForce = 10.0f;

        public Vector2 Velocity = new Vector2();
        private RigidBody2DComponent m_PhysicsBody;
        private Material m_MeshMaterial;

        private int m_CollisionCounter = 0;

        public Vector2 MaxSpeed = new Vector2();

        private bool Colliding => m_CollisionCounter > 0;

        void OnCreate()
        {
            m_PhysicsBody = GetComponent<RigidBody2DComponent>();

            MeshComponent meshComponent = GetComponent<MeshComponent>();
            m_MeshMaterial = meshComponent.Mesh.GetMaterial(0);
            m_MeshMaterial.Set("u_Metalness", 0.0f);
        }

        public void OnCollisionBegin(float data)
        {
            m_CollisionCounter++;
        }

        public void OnCollisionEnd(float data)
        {
            m_CollisionCounter--;
        }

        void OnFixedUpdate()
        {
            float ts = Time.FixedDeltaTime;
            float movementForce = HorizontalForce;
            Velocity = m_PhysicsBody.LinearVelocity;
            if (!Colliding)
            {
                movementForce *= 0.4f;
            }

            if (Input.IsKeyPressed(KeyCode.D))
                m_PhysicsBody.ApplyLinearImpulse(new Vector2(movementForce, 0), new Vector2(), true);
            else if (Input.IsKeyPressed(KeyCode.A))
                m_PhysicsBody.ApplyLinearImpulse(new Vector2(-movementForce, 0), new Vector2(), true);

            if (Colliding && Input.IsKeyPressed(KeyCode.Space))
                m_PhysicsBody.ApplyLinearImpulse(new Vector2(0, JumpForce), new Vector2(0, 0), true);

            if (m_CollisionCounter > 0)
                m_MeshMaterial.Set("u_AlbedoColor", new Vector3(1.0f, 0.0f, 0.0f));
            else
                m_MeshMaterial.Set("u_AlbedoColor", new Vector3(0.8f, 0.8f, 0.8f));

        }

    }
}

File: PrismEditor/Assets/scripts/Python/PlayerCube.py
from Prism import (
    Behaviour, Input, Time, KeyCodes,
    RigidBody2DComponent, MeshRendererComponent,
)
from Prism.Math import Vector2, Vector3
from Prism.Renderer.Material import Material


class PlayerCube(Behaviour):
    HorizontalForce: float = 10.0
    JumpForce: float = 10.0
    Velocity: Vector2 = Vector2(0, 0)
    MaxSpeed: Vector2 = Vector2(0, 0)

    def OnCreate(self):
        self.m_PhysicsBody = self.GetComponent(RigidBody2DComponent)

        meshComponent = self.GetComponent(MeshRendererComponent)
        self.m_MeshMaterial = meshComponent.GetMaterial(0)
        self.m_MeshMaterial.Set("u_Metalness", 0.0)

        self.m_CollisionCounter = 0

    def OnCollisionBegin(self, data):
        self.m_CollisionCounter += 1

    def OnCollisionEnd(self, data):
        self.m_CollisionCounter -= 1

    @property
    def _colliding(self):
        return self.m_CollisionCounter > 0

    def OnFixedUpdate(self):
        ts = Time.FixedDeltaTime
        movementForce = self.HorizontalForce
        self.Velocity = self.m_PhysicsBody.LinearVelocity
        if not self._colliding:
            movementForce *= 0.4

        if Input.IsKeyPressed(KeyCodes.D):
            self.m_PhysicsBody.ApplyLinearImpulse(Vector2(movementForce, 0), Vector2(), True)
        elif Input.IsKeyPressed(KeyCodes.A):
            self.m_PhysicsBody.ApplyLinearImpulse(Vector2(-movementForce, 0), Vector2(), True)

        if self._colliding and Input.IsKeyPressed(KeyCodes.Space):
            self.m_PhysicsBody.ApplyLinearImpulse(Vector2(0, self.JumpForce), Vector2(0, 0), True)

        if self.m_CollisionCounter > 0:
            self.m_MeshMaterial.Set("u_AlbedoColor", Vector3(1.0, 0.0, 0.0))
        else:
            self.m_MeshMaterial.Set("u_AlbedoColor", Vector3(0.8, 0.8, 0.8))
